test/ChildChainGaugeInjectorV2.js

```diff
@@ -0,0 +1,372 @@
+        await hre.ethers.provider.send("evm_mine");
+        console.log(await currentChainTime());
+
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        expect(upkeepNeeded).to.equal(true);
+
+        const initialBalance = await token.balanceOf(gaugeAddress);
+        await injector.performUpkeep(performData);
+        expect(await token.balanceOf(gaugeAddress) - initialBalance).to.equal(weeklyIncentive);
+
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        expect(upkeepNeeded).to.equal(false);
+
+        await hre.ethers.provider.send("evm_increaseTime", [60 * 60 * 24 * 8]);
+        await hre.ethers.provider.send("evm_mine");
+        console.log(await currentChainTime());
+
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        expect(upkeepNeeded).to.equal(false);
+    });
+
+    it("should not run too soon", async function () {
+        const tokenAddress = await token.getAddress();
+        const gaugeAddress = await gauge.getAddress();
+        const injectorAddress = await injector.getAddress();
+
+        const weeklyIncentive = BigInt("200000000000000000000");
+        await injector.addRecipients([gauge], weeklyIncentive, 2, 0)
+
+        await token.transfer(injectorAddress, weeklyIncentive * 2n)
+        let rewardData = await gauge.reward_data(tokenAddress);
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        if (!upkeepNeeded) {
+            const sleepTime = BigInt(rewardData[1]) - BigInt(await currentChainTime());
+            await hre.ethers.provider.send("evm_increaseTime", [sleepTime.toString()]);
+            await hre.ethers.provider.send("evm_mine");
+            const {upkeepNeeded, performData} = await injector.checkUpkeep("0x");
+            expect(upkeepNeeded).to.equal(true);
+        }
+        expect(await token.balanceOf(injectorAddress)).to.greaterThanOrEqual(weeklyIncentive)
+        await injector.performUpkeep(performData);
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        rewardData = await gauge.reward_data(tokenAddress);
+        [distributor, periodFinished, rate, lastUpdated, integral] = rewardData;
+        let sleepTime = Math.floor(Math.random() * 60 * 60 * 24 * 6) + 1;
+        await hre.ethers.provider.send("evm_increaseTime", [sleepTime]);
+        await hre.ethers.provider.send("evm_mine");
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        expect(upkeepNeeded).to.equal(false);
+
+        sleepTime = BigInt(rewardData[1]) - BigInt(await currentChainTime());
+        await hre.ethers.provider.send("evm_increaseTime",[sleepTime.toString()]);
+        await hre.ethers.provider.send("evm_mine");
+        [upkeepNeeded, performData] = await injector.checkUpkeep("0x");
+        expect(upkeepNeeded).to.equal(true);
+    });
+
+    it("should work with a long delay", async function () {
```

Comment:
Can you explain what these two tests are doing/testing.  What are the long and short delays that should cause it to not work?

Reply:
This is a test taken from the old brownie suite. It tests if the upkeep is still possible after some amount of time and works as intendend